lib/extract/netcdf/extract_netcdf_base.jinja

```diff
@@ -0,0 +1,464 @@
+        ! variables that are going to be declared:
+        allocate(this%var_id(num_pre_vars+num_post_vars))
+        
+    end subroutine PreStart
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine is called to open a NetCDF file for reading. The
+    !! filename is based on the module and kernel name. This is used by a
+    !! driver program that will read a NetCDF file previously created by the
+    !! PSyData API.
+    !! @param[inout] this The instance of the ExtractNetcdfBaseType.
+    !! @param[in] module_name The name of the module of the instrumented
+    !!            region.
+    !! @param[in] kernel_name The name of the instrumented region.
+    subroutine OpenRead(this, module_name, kernel_name)
+        use netcdf, only : nf90_open, NF90_NOWRITE
+        implicit none
+        class(ExtractNetcdfBaseType), intent(inout), target :: this
+        character(*), intent(in) :: module_name, kernel_name
+        integer :: retval
+
+        retval = CheckError(nf90_open(module_name//"-"//kernel_name//".nc", &
+                                        NF90_NOWRITE, this%ncid))
+    end subroutine OpenRead
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine is called once all variables are declared (this includes
+    !! variables that will be written before as well as variables that are
+    !! written after the instrumented region). It is used to switch the NetCDF
+    !! file from declaration to writing state, and reset the next_var_index
+    !! back to 1.
+    !! @param[inout] this The instance of the ExtractNetcdfBaseType.
+    subroutine PreEndDeclaration(this)
+        use netcdf, only : nf90_enddef
+        implicit none
+        class(ExtractNetcdfBaseType), intent(inout), target :: this
+        integer :: retval
+
+        call this%PSyDataBaseType%PreEndDeclaration()
+        retval = CheckError(nf90_enddef(this%ncid))
+    end subroutine PreEndDeclaration
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine is called after the value of all variables has been
+    !! provided (and declared). After this call the instrumented region will
+    !! be executed. For this NetCDF interface this function is not required,
+    !! so it is empty.
+    !! @param[inout] this The instance of the ExtractNetcdfBaseType.
+    subroutine PreEnd(this)
+        implicit none
+        class(ExtractNetcdfBaseType), intent(inout), target :: this
+        call this%PSyDataBaseType%PreEnd()
+    end subroutine PreEnd
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine is called after the instrumented region has been
+    !! executed. After this call the value of variables after the instrumented
+    !! region will be provided. For this NetCDF interface this function is
+    !! not required, so it is empty.
```

Comment:
Again, it's not strictly empty. It feels like here (and above) we're overriding a routine, only to call the base implementation. However, I can see it's not quite that simple as we have `this%PSyDataBaseType%PostStart()` rather than just `this%PostStart()`?

Reply:
This should also be considered empty ... and it is gone. But: ``this%PostStart()`` would call itself recursively, you have to specify the base class here (afaik).